dev/tools/controllerbuilder/pkg/commands/generatetypes/generatetypescommand.go

```diff
@@ -189,6 +189,14 @@ func RunGenerateCRD(ctx context.Context, o *GenerateCRDOptions) error {
 				return fmt.Errorf("add refs file %s: %w", scaffolder.PathToRefsFile(kind, resource.ProtoName), err)
 			}
 		}
+		if scaffolder.IdentityFileExist(kind, resource.ProtoName) {
```

Comment:
Not sure it needs to be another file, but ... we can see!

Reply:
I grouped them by logical unit. The identity file is for the resource own usage. The reference file is for other resources. For contributors using reference, they don't need to know and read the identity logic.